Documentation/Diagnostics/PH2006.md

```diff
@@ -0,0 +1,73 @@
+# PH2006: Namespace matches File Path
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2006 |
+| Category  | [Naming](../Naming.md) |
+| Analyzer | [NamespaceMatchFilePathAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/Naming/NamespaceMatchFilePathAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+In order to prevent pollution of namespaces, the File Path and Namespace must match. There are two modes.
+
+### Path contains folder that matches the Namespace (Default)
+
+In the default configuration, the path must include a folder that exactly matches the namespace.
+
+### Path ends with Namespace
+
+In this more strict mode, the namespace must exactly match the end of the file path. For the purposes of comparison, path separators are treated as dots in the namespace. 
+
+## How to solve
+
+Ideally, use the more strict, non-default, configuration. Then consistently have the namespaces of all of your files match the folder structure.
+
+## Exceptions
+
+The following namespaces are always exempt from this rule. See supporting information for reasoning.
+
+| Namespace | Justification  |
+|--|--|
+| `System.Runtime.CompilerServices` | [Init setter support in .NET Standard 2.x](https://developercommunity.visualstudio.com/t/error-cs0518-predefined-type-systemruntimecompiler/1244809) |
+
+## Example
+
+If the file path is `.\Philips.CodeAnalysis.MaintainabilityAnalyzers\Naming\NamespaceMatchFilePathAnalyzer.cs` then the following code will pass for the default mode but fail in the more strict mode:
```

Comment:
I prefer to use obvious example names, as opposed to values that _could_ be confused with real names in the code.

Reply:
@ynsehoornenborg I don't understand can you elaborate?

Reply:
I think I understand. Updated example.